### PR TITLE
Migrate post view module to typescript 

### PR DESCRIPTION
**Pull request by sridhar02**
#### Summary

Conver all files related to post view to typescript

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/15445

### CODE REVIEW

File: components/post_view/post_view.test.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import React from 'react';
 import {shallow} from 'enzyme';
 
 import PostList from './post_list';
-import PostView from './post_view.jsx';
+import PostView from './post_view.js';
```
Comment (jespino) [inline components/post_view/post_view.test.tsx:8]: I think this should be just `'./post_view;'`

**Review by jespino — CHANGES_REQUESTED** [review]
I tiny change needed, but otherwise, looks good to me.

File: components/post_view/index.ts [hunk 2]
```diff
@@ -34,8 +46,26 @@ export const isChannelLoading = (params, channel, team, teammate, teamMembership
...
+    channelId: string;
+    focusedPostId: string;
+    match: {
+        params: {
+            team: string;
+        };
+    };
+}
+
+type State ={
```
Comment (lieut-data) [inline components/post_view/index.ts:61]: nit: Is there a missing space after `=` here?

File: components/post_view/index.ts [hunk 3]
```diff
@@ -6,12 +6,24 @@ import {withRouter} from 'react-router-dom';
...
+import PostView from './post_view.js';
+
+type Params = {
+    identifier: string;
+    team: string;
+    postid?: string;
+    path: string;
+}
+
+export const isChannelLoading = (params: Params, channel: Channel, team: Team, teammate: UserProfile, teamMemberships:TeamMembership) => {
```
Comment (lieut-data) [inline components/post_view/index.ts:26]: nit: Is there a missing space after `teamMemberships:` here?

File: components/post_view/post_view.tsx [hunk 4]
```diff
@@ -8,15 +8,28 @@ import LoadingScreen from 'components/loading_screen';
...
 
-export default class PostView extends React.PureComponent {
+type Props = {
+    lastViewedAt: number;
+    channelLoading: boolean;
+    channelId: string;
+    focusedPostId: string;
+}
+
+type State ={
```
Comment (lieut-data) [inline components/post_view/post_view.tsx:18]: nit: Is there a missing space after `=` here?

File: components/post_view/post_view.tsx [hunk 5]
```diff
@@ -41,18 +54,21 @@ export default class PostView extends React.PureComponent {
...
-    changeUnreadChunkTimeStamp = (unreadChunkTimeStamp) => {
-        this.setState({
-            unreadChunkTimeStamp,
-            loaderForChangeOfPostsChunk: true,
-        }, () => {
-            window.requestAnimationFrame(() => {
-                this.setState({
-                    loaderForChangeOfPostsChunk: false,
+    changeUnreadChunkTimeStamp = (unreadChunkTimeStamp: number) => {
+        this.setState(
```
Comment (lieut-data) [inline components/post_view/post_view.tsx:58]: Are the whitespace changes here required? Subjectively, the previous formatting seemed more familiar within this codebase.

**Review by lieut-data — APPROVED** [review]
Looks good, just a few formatting questions.

**Comment by jfrerich** [comment]
@sridhar02, can you address the PR provided feedback? Also, there is a failing build. You can find these failures [here](https://app.circleci.com/pipelines/github/mattermost/mattermost-webapp/29557/workflows/54b7c33e-4b1a-4e23-81ca-a47a4716c125/jobs/195534) or on the `Details` link next to the failing checks.  Those will need to be passing with a green check before we can merge.  Thanks!

**Comment by sridhar02** [comment]
@jfrerich I was busy with some other work I will update the PR in few days.

**Comment by devinbinnie** [comment]
@sridhar02 Are you still planning on finishing up this PR?

**Comment by sridhar02** [comment]
@devinbinnie not getting enough time on this I am unassigning myself from the issue.

**Comment by devinbinnie** [comment]
Closing this PR and re-opening the issue.

**Pull request closed without merging**